readme.md

```diff
@@ -541,7 +541,330 @@ namespace Azure.Service.Models
+        }
+
+        internal static Cat DeserializeCat(JsonElement element)
+        {
+            if (element.ValueKind == JsonValueKind.Null)
+            {
+                return null;
+            }
+            string name = default;
+            foreach (var property in element.EnumerateObject())
+            {
+-               if (property.NameEquals("name"u8))
++               if (property.NameEquals("catName"u8))
+                {
+                    meow = property.Value.GetString();
+                    continue;
+                }
+            }
+            return new Cat(name);
+        }
+    }
+}
+```
+
+You can also add multiple element in the attribute and the generator will generate the property in a hierarchy.
+
+**NOTE: Introducing extra layers in serialized JSON only works for mgmt plane and HLC models, does not work for DPG models.**
+
+``` C#
+public partial class Cat
+{
+    [CodeGenMemberSerialization("properties", "catName")]
+    public string Name { get; set; }
+}
+```
+
+**Generated code after (Generated/Models/Model.cs):**
+
+``` diff
+namespace Azure.Service.Models
+{
+    public partial class Cat
+    {
+-        public string Name { get; set; }  
+    }
+}
+```
+
+**Generated code after (Generated/Models/Model.Serialization.cs):**
+
+``` diff
+namespace Azure.Service.Models
+{
+    public partial class Cat : IUtf8JsonSerializable
+    {
+        void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+-           writer.WritePropertyName("name"u8);
++           writer.WritePropertyName("properties"u8);
++           writer.WriteStartObject();
++           writer.WritePropertyName("catName"u8);
+            writer.WriteStringValue(Name);
++           writer.WriteEndObject();
+            writer.WriteEndObject();
+        }
+
+        internal static Cat DeserializeCat(JsonElement element)
+        {
+            if (element.ValueKind == JsonValueKind.Null)
+            {
+                return null;
+            }
+            string name = default;
+            foreach (var property in element.EnumerateObject())
+            {
+-               if (property.NameEquals("name"u8))
++               if (property.NameEquals("properties"u8))
++               {
++                   foreach (var property in element.EnumerateObject())
++                   {
++                       if (property.NameEquals("catName"u8))
+                        {
+                            meow = property.Value.GetString();
+                            continue;
+                        }
++                   }
++                   continue;
++               }
+            }
+            return new Cat(name);
+        }
+    }
+}
+```
+
+</details>
+
+If you want to change the implementation of serialization/deserialization method of one particular property, you could define your own hook methods and assign them to a property using the `CodeGenMemberSerializationHooks` attribute.
```

Comment:
Maybe It is better to separate it as another customization.

Reply:
I believe this is also one of the topics of "customization serialization/deserialization method". I did not have a better name that distinguishing this from the topic above, and this attribute can be used together with the attribute above. Therefore I do not want this to be in another section

Reply:
Resolved in another PR